processing/src/main/java/io/druid/output/OutputMedium.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.output;
+
+import io.druid.java.util.common.io.Closer;
+
+import java.io.Closeable;
+import java.io.IOException;
+
+/**
+ * OutputMedium manages the resources of a bunch of {@link OutputBytes} objects of the same kind, created by calling
+ * {@link #makeOutputBytes()} on the OutputMedium object. When OutputMedium is closed, all child OutputBytes couldn't
+ * be used anymore.
+ */
+public interface OutputMedium extends Closeable
```

Comment:
this is called `OutputMedium` but it is actually used to do Input/Output thus calling it OutputMedium might be misleading, I will recommend keeping the same name as IOpeon or something that has IO.

Reply:
I don't want "IO" because it's too generic. Reading case of `OutputBytes` is quite special, and actually this abstraction is used only during creating segments, and not expected to have more use cases in the future. It would probably be more distinctive to call it "SegmentWriteOutMedium" and "WriteOutBytes", what do you think?

Reply:
@b-slim could you please comment here?

Reply:
Sorry, I guess `SegmentWriteOutMedium` is good one.

Reply:
Renamed